jsyntrax/src/main/java/org/atpfivt/jsyntrax/styles/MatchException.java

```diff
@@ -0,0 +1,7 @@
+package org.atpfivt.jsyntrax.styles;
+
+public class MatchException extends Throwable {
```

Comment:
First, you should **never** extend `Throwable` directly. See my lecture no. 4.

Second, you should rarely invent your own exception classes. There are plenty of standard ones, e. g. `InvalidStateException` can be used here. I think here is no need for a specialized exception class